packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -358,19 +351,11 @@ function computeShadowMode(vm: VM) {
                 if (def.shadowSupportMode === ShadowSupportMode.Any) {
                     shadowMode = ShadowMode.Native;
                 } else {
-                    const shadowAncestor = getNearestShadowAncestor(vm);
-                    if (
-                        !isNull(shadowAncestor) &&
-                        shadowAncestor.shadowMode === ShadowMode.Native
-                    ) {
-                        // Transitive support for native Shadow DOM. A component in native mode
-                        // transitively opts all of its descendants into native.
-                        shadowMode = ShadowMode.Native;
-                    } else {
-                        // Synthetic if neither this component nor any of its ancestors are configured
-                        // to be native.
-                        shadowMode = ShadowMode.Synthetic;
-                    }
+                    // Transitive support for native Shadow DOM. A component in native mode
+                    // transitively opts all of its descendants into native.
+                    // Synthetic if neither this component nor any of its ancestors are configured
+                    // to be native.
+                    shadowMode = owner?.nearestSRMode || ShadowMode.Synthetic;
```

Comment:
```suggestion
                    shadowMode = owner?.nearestSRMode ?? ShadowMode.Synthetic;
```

the mode may be `0`, in which case `||` will cause it to fall back to `ShadowMode.Synthetic`.

This fixes the Karma test failures for me locally.